src/g_game.c

```diff
@@ -369,93 +368,101 @@ static void G_DemoSkipTics(void)
-  }
-  else
-  {
-    cmd_prev_carry.pitch = 0.0;
-    cmd_carry.pitch = 0.0;
-    return 0;
-  }
+  return CarryError(pitch, &prevcarry.pitch, &carry.pitch);
 }
 
-static int CalcMouseStrafe(int mousex)
+static int CarryMouseVert(double vert)
 {
-  if (mouseSensitivity_horiz_strafe)
-  {
-    const double desired = (cmd_prev_carry.strafe + I_AccelerateMouse(mousex) *
-                            (mouseSensitivity_horiz_strafe + 5) * 2 / 10);
-    const int actual = lround(desired * 0.5) * 2; // Even values only.
-    cmd_carry.strafe = desired - actual;
-    return actual;
-  }
-  else
-  {
-    cmd_prev_carry.strafe = 0.0;
-    cmd_carry.strafe = 0.0;
-    return 0;
-  }
+  return CarryError(vert, &prevcarry.vert, &carry.vert);
 }
 
-static int CalcMouseVert(int mousey)
+static int CarryMouseSide(double side)
 {
-  if (mouseSensitivity_vert)
-  {
-    const double vert = (I_AccelerateMouse(mousey) *
-                         (mouseSensitivity_vert + 5) / 10);
-    return CarryError(vert, &cmd_prev_carry.vert, &cmd_carry.vert);
-  }
-  else
-  {
-    cmd_prev_carry.vert = 0.0;
-    cmd_carry.vert = 0.0;
-    return 0;
-  }
+  const double desired = side + prevcarry.side;
+  const int actual = lround(desired * 0.5) * 2; // Even values only.
+  carry.side = desired - actual;
+  return actual;
 }
 
-void G_MouseMovementResponder(const event_t *ev)
+static short CarryLowResAngle(short angle)
 {
-  if (strictmode && demorecording && skip_mouse)
-    return;
+  const short desired = angle + prevcarry.lowres;
+  // Round to nearest 256 for single byte turning. From Chocolate Doom.
+  const short actual = (desired + 128) & 0xFF00;
+  carry.lowres = desired - actual;
+  return actual;
+}
+
+static double CalcMouseAngle(int mousex)
+{
+  if (!mouseSensitivity_horiz)
+    return 0.0;
+
+  return (I_AccelerateMouse(mousex) * (mouseSensitivity_horiz + 5) * 8 / 10);
+}
+
+static double CalcMousePitch(int mousey)
+{
+  if (!mouseSensitivity_vert_look)
+    return 0.0;
+
+  return (I_AccelerateMouse(mousey) * direction[mouse_y_invert] *
+          (mouseSensitivity_vert_look + 5) / 10);
+}
+
+static double CalcMouseSide(int mousex)
+{
+  if (!mouseSensitivity_horiz_strafe)
+    return 0.0;
+
+  return (I_AccelerateMouse(mousex) *
+          (mouseSensitivity_horiz_strafe + 5) * 2 / 10);
+}
+
+static double CalcMouseVert(int mousey)
+{
+  if (!mouseSensitivity_vert)
+    return 0.0;
+
+  return (I_AccelerateMouse(mousey) * (mouseSensitivity_vert + 5) / 10);
+}
 
-  mousex += ev->data2;
-  mousey += ev->data3;
+void G_PrepTiccmd(void)
+{
+  ticcmd_t *cmd = &basecmd;
 
   if (!M_InputGameActive(input_strafe))
-    localview.angle = CalcMouseAngle(mousex);
+  {
+    localview.rawangle = -CalcMouseAngle(mousex);
+    cmd->angleturn = CarryAngle(localview.rawangle);
+    if (lowres_turn)
+    {
+      cmd->angleturn = CarryLowResAngle(cmd->angleturn);
+    }
+    localview.angle = cmd->angleturn << 16;
```

Comment:
Is this `angle_t`? I suggest using `<< FRACBITS`.

Reply:
I used `FRACBITS` before, but then I realized it's not converting the angle to `fixed_t`, it's just scaling up the value to `angle_t` and `<< 16` is just a coincidence. I'm following this step exactly: https://github.com/fabiangreffrath/woof/blob/b2c7b4064103c45d73ee76c4a740da6246563c24/src/p_user.c#L190

Reply:
Got it, thanks.